business/openshift_oauth.go

```diff
@@ -0,0 +1,194 @@
+package business
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+
+	kube "k8s.io/client-go/kubernetes"
+)
+
+type OpenshiftOAuthService struct {
+	k8s kubernetes.IstioClientInterface
+}
+
+type OAuthMetadata struct {
+	AuthorizationEndpoint string `json:"authorizationEndpoint"`
+}
+
+// Structure that's returned by the openshift oauth authorization server.
+// It defaults to following the snake_case format, so we parse it to something
+// more usable on our side.
+type OAuthAuthorizationServer struct {
+	AuthorizationEndpoint string `json:"authorization_endpoint"`
+}
+
+type OAuthUser struct {
+	Metadata OAuthUserMetadata `json:"metadata"`
+}
+
+type OAuthUserMetadata struct {
+	Name string `json:"name"`
+}
+
+type OAuthRoute struct {
+	Spec OAuthRouteSpec `json:"spec"`
+}
+
+type OAuthRouteSpec struct {
+	Host string             `json:"host"`
+	TLS  *OAuthRouteTLSSpec `json:"tls,omitempty"`
+}
+
+type OAuthRouteTLSSpec struct {
+	Termination string `json:"termination"`
+}
+
+const serverPrefix = "https://openshift.default.svc/"
+
+func (in *OpenshiftOAuthService) Metadata() (metadata *OAuthMetadata, err error) {
+	var server *OAuthAuthorizationServer
+
+	if err != nil {
+		message := fmt.Errorf("Could not parse data from the Openshift API: %v", err)
+		return nil, message
+	}
+
+	response, err := request("GET", ".well-known/oauth-authorization-server", nil)
+
+	if err != nil {
+		message := fmt.Errorf("Could not parse data from the Openshift API: %v", err)
+		return nil, message
+	}
+
+	err = json.Unmarshal(response, &server)
+
+	if err != nil {
+		message := fmt.Errorf("Could not parse data from the Openshift API: %v", err)
+		return nil, message
+	}
+
+	redirectURL, err := getKialiRoutePath()
```

Comment:
shouldn't you have another if-stmt here to check if err != nil and if so return an error message like the other if-stmt's above that check err status?